dspace-api/src/main/java/org/dspace/handle/HandlePlugin.java

```diff
@@ -432,4 +428,121 @@ public Enumeration getHandlesForNA(byte[] theNAHandle)
             }
         }
     }
+
+    public static Map<String, String> extractMetadata(DSpaceObject dso) {
+        Map<String, String> map = new LinkedHashMap<>();
+        if (null != dso) {
+            List<Metadatum> mds = dso.getMetadata("dc", "title", null, Item.ANY, null);
+            if (0 < mds.size()) {
+                map.put(AbstractPIDService.HANDLE_FIELDS.TITLE.toString(), mds.get(0).value);
+            }
+            map.put(AbstractPIDService.HANDLE_FIELDS.REPOSITORY.toString(), repositoryName);
+            mds = dso.getMetadata("dc", "date", "accessioned", Item.ANY, null);
+            if (0 < mds.size()) {
+                map.put(AbstractPIDService.HANDLE_FIELDS.SUBMITDATE.toString(), mds.get(0).value);
+            }
+            map.put(AbstractPIDService.HANDLE_FIELDS.REPORTEMAIL.toString(), repositoryEmail);
+        }
+        return map;
+    }
 }
+
+class ResolvedHandle {
+    List<HandleValue> values;
+    private int idx = -1;
+
+    public ResolvedHandle(String url, String title, String repository, String submitdate) {
+        init(url, title, repository, submitdate);
+    }
+
+    public ResolvedHandle(String url, DSpaceObject dso) {
+        String title = null;
+        String repository = null;
+        String submitdate = null;
+        String reportemail = null;
+        if (null != dso) {
+            Map<String, String> map = HandlePlugin.extractMetadata(dso);
+            String key
+                    = AbstractPIDService.HANDLE_FIELDS.TITLE.toString();
+            title = map.getOrDefault(key, "");
+
+            key = AbstractPIDService.HANDLE_FIELDS.REPOSITORY.toString();
+            repository = map.getOrDefault(key, "");
+
+            key = AbstractPIDService.HANDLE_FIELDS.SUBMITDATE.toString();
+            submitdate = map.getOrDefault(key, "");
+        }
+        init(url, title, repository, submitdate, reportemail);
```

Comment:
see the branch rather than this PR https://github.com/ufal/lindat-dspace/compare/lindat...ufal:issue_231